### PR TITLE
feat: tests added.

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**

- added some tests to the project.

### CODE REVIEW

**Comment by KhudaDad414** [comment]
@magicmatatjahu @derberg
Both of the PRs are ready for review.

File: test/ComponentProvider.spec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  test('should not contain any component from components section', () => {
+    for (const key of componentProvider.parameters.keys()) {
+      expect(isInComponents(key)).toBe(false);
+    }
+    for (const key of componentProvider.messages.keys()) {
+      expect(isInComponents(key)).toBe(false);
+    }
+    for (const key of componentProvider.schemas.keys()) {
+      expect(isInComponents(key)).toBe(false);
+    }
```
Comment (magicmatatjahu) [inline test/ComponentProvider.spec.ts:21]: Please also check the Yaml with components.

**Review by magicmatatjahu — COMMENTED** [review]
Few comments :) Overall good.

File: package.json [hunk 2]
```diff
@@ -39,10 +40,10 @@
     "eslint-plugin-github": "^4.1.3",
     "eslint-plugin-security": "^1.4.0",
     "eslint-plugin-sonarjs": "^0.8.0-125",
-    "jest": "^27.0.4",
+    "jest": "^27.0.6",
     "semantic-release": "^17.4.4",
-    "typescript": "^4.3.2",
-    "conventional-changelog-conventionalcommits": "^4.6.0"
```
[the remark this answers is not in the document]
Comment (KhudaDad414) [reply package.json:45]: readded it.

**Comment by magicmatatjahu** [comment]
@KhudaDad414 Please resolve conflicts :)

**Comment by derberg** [comment]
@KhudaDad414 I don't think you resolved conflicts well as changes like adding `"docs": "jsdoc2md lib/Optimizer.js -f lib/**/*.js > API.md",` should not show up in your PR as you did it in the other one that is already merged

**Comment by KhudaDad414** [comment]
> @KhudaDad414 I don't think you resolved conflicts well as changes like adding `"docs": "jsdoc2md lib/Optimizer.js -f lib/**/*.js > API.md",` should not show up in your PR as you did it in the other one that is already merged

I can resolve the conflicts of `package.json` but the `package-lock.json` has more than 1200 conflicts. I searched for a way to resolve it's conflicts but I'm out of luck. :(

Any suggestions?

**Comment by derberg** [comment]
package-lock is a beast but you can do it this way:
- solve conflicts by accepting everything that comes from master
- run `npm i` and on top of what you merged, you will create another commit that adds your packages to the lock file

**Comment by KhudaDad414** [comment]
@derberg I think I made it worse.
Good news:
The conflict is gone.
Bad news:
now it shows all of the previous PRs changes here.
can't find jest (although it works fine locally)

for the solution, I've opened a new PR. 
The previous PR changes are gone. but still, PR testing fails. :(

**Comment by derberg** [comment]
@KhudaDad414 tbh, always when I read the state you reached, I tend to just give up and create new branch that is already up to date with master and apply my changes there and then open new PR 🤷🏼

**Pull request closed without merging**